Makefile

```diff
@@ -43,98 +41,73 @@ DEV_MOUNTS = '-v $(CWD)/$(FV3):/$(FV3)/$(FV3) -v $(CWD)/tests:/$(FV3)/tests -v $
 clean:
 	find . -name ""
 	$(RM) -rf comparison/wrapped/output/*
+	$(MAKE) -C external/fv3gfs-wrapper clean
+	$(MAKE) -C external/fv3gfs-fortran clean
 
 update_submodules:
 	if [ ! -f $(FV3UTIL_DIR)/requirements.txt  ]; then \
 		git submodule update --init --recursive; \
 	fi
 
+constraints.txt: requirements.txt requirements_wrapper.txt requirements_lint.txt
+	pip-compile $^ --output-file constraints.txt
 
-build_environment:
-	DOCKER_BUILDKIT=1 docker build \
-		--network host \
-		--build-arg MIDBASE=$(BASE_INSTALL) \
-		-f $(CWD)/docker/Dockerfile.build_environment \
-		-t $(FV3_INSTALL_IMAGE) \
-		--target $(FV3_INSTALL_TARGET) \
-		.
-
-build_wrapped_environment:
-	DOCKER_BUILDKIT=1 docker build \
-		--network host \
-		-f $(CWD)/docker/Dockerfile.build_environment \
-		-t $(WRAPPER_INSTALL_IMAGE) \
-		--target $(FV3_INSTALL_TARGET) \
-		--build-arg BASE_IMAGE=$(WRAPPER_IMAGE) \
-		.
+# Image build instructions have moved to docker/Makefile but are kept here for backwards-compatibility
+
+build_environment: update_submodules
+	$(MAKE) -C docker build_core_deps
+
+build_wrapped_environment: update_submodules
+	$(MAKE) -C docker build_deps
 
 build: update_submodules
 	if [ $(PULL) == True ]; then \
 		$(MAKE) pull_environment_if_needed; \
 	else \
 		$(MAKE) build_environment; \
 	fi
-	docker build \
-		--network host \
-		--build-arg build_image=$(FV3_INSTALL_IMAGE) \
-		-f $(CWD)/docker/Dockerfile \
-		-t $(FV3_IMAGE) \
-		.
+	$(MAKE) -C docker fv3core_image
 
 build_wrapped: update_submodules
 	if [ $(PULL) == True ]; then \
 		$(MAKE) pull_wrapped_environment_if_needed; \
 	else \
-		$(MAKE) build_environment; \
-	$(MAKE) build_wrapped_environment
-	docker build \
-		--network host \
-		--no-cache \
-		--build-arg build_image=$(WRAPPER_INSTALL_IMAGE) \
-		-f $(CWD)/docker/Dockerfile \
-		-t $(WRAPPED_FV3_IMAGE) \
-		.
-
+		$(MAKE) build_wrapped_environment; \
+	fi
+	$(MAKE) -C docker fv3core_wrapper_image
 
 pull_environment_if_needed:
-	if [ -z $(shell docker images -q $(FV3_INSTALL_IMAGE)) ]; then \
-		docker pull $(FV3_INSTALL_IMAGE); \
-	fi
+	$(MAKE) -C docker pull_core_deps_if_needed
 
 pull_wrapped_environment_if_needed:
```

Comment:
this one now has an error:
/bin/bash: line 0: [: too many arguments ...

Reply:
Fixed, thanks!